src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -176,18 +176,17 @@ impl<F: FieldExt> CondSwapChip<F> {
             let b_swapped = meta.query_advice(config.b_swapped, Rotation::cur());
             let swap = meta.query_advice(config.swap, Rotation::cur());
 
-            let one = Expression::Constant(F::one());
-
-            // a_swapped - b ⋅ swap - a ⋅ (1-swap) = 0
             // This checks that `a_swapped` is equal to `b` when `swap` is set,
             // but remains as `a` when `swap` is not set.
-            let a_check =
-                a_swapped - b.clone() * swap.clone() - a.clone() * (one.clone() - swap.clone());
+            let a_check = ternary(
+                swap.clone(),
+                a_swapped.clone() - b.clone(),
+                a_swapped - a.clone(),
+            );
 
-            // b_swapped - a ⋅ swap - b ⋅ (1-swap) = 0
             // This checks that `b_swapped` is equal to `a` when `swap` is set,
             // but remains as `b` when `swap` is not set.
-            let b_check = b_swapped - a * swap.clone() - b * (one - swap.clone());
+            let b_check = ternary(swap.clone(), b_swapped.clone() - a, b_swapped - b);
```

Comment:
Similarly, this would have been simpler as:
```suggestion
            let b_check = b_swapped - ternary(swap.clone(), a, b);
```
Similarly weakly blocking: if we take the encompassing approach, it would be nice to make the change across two commits, so the diff on `circuit_description` is easier to review.